### PR TITLE
add more settings to strategy

### PR DESCRIPTION
**Pull request by xmatthias**
## Summary
Add experimental settings to strategy - as it's almost exclusively strategy-dependant if the sell-signal makes sense or not.

* Drop mandatory column in the documentation hoping that this will make this table simpler to read.

### CODE REVIEW

File: freqtrade/resolvers/strategy_resolver.py [hunk 1]
```diff
@@ -38,24 +38,38 @@ def __init__(self, config: Optional[Dict] = None) -> None:
...
+                      ("process_only_new_candles",      None, False),
+                      ("order_types",                   None, False),
+                      ("order_time_in_force",           None, False),
+                      ("use_sell_signal",               False, True),
+                      ("sell_profit_only",              False, True),
+                      ("ignore_roi_if_buy_signal",      False, True),
                       ]
-        for attribute in attributes:
-            self._override_attribute_helper(config, attribute)
+        for attribute, default, experimental in attributes:
```
Comment (shusso) [inline freqtrade/resolvers/strategy_resolver.py]: nitpicking, these could be extracted to it's own methods e.g. `override_attributes` and `log_strategy_attributes`

File: freqtrade/resolvers/strategy_resolver.py [hunk 2]
```diff
@@ -66,13 +80,24 @@ def __init__(self, config: Optional[Dict] = None) -> None:
 
         self._strategy_sanity_validations()
 
-    def _override_attribute_helper(self, config, attribute: str):
+    def _override_attribute_helper(self, config, attribute: str, default):
```
Comment (shusso) [inline freqtrade/resolvers/strategy_resolver.py]: should `default` be `None` by default?

Comment (xmatthias) [reply freqtrade/resolvers/strategy_resolver.py]: there is not really a point tbh - it's always passed in within the loop and it's not used anywhere else ...

File: freqtrade/resolvers/strategy_resolver.py (hunk 1 above)
Comment (mishaker) [reply freqtrade/resolvers/strategy_resolver.py]: @shusso: Not a fan honestly :/ it is not that big and it is already encapsulated in strategy_resolver ...

**Comment by mishaker** [comment]
@shusso I waited for you to accept your requested changed but I trust you are satisfied with the result.

**Pull request merged**